vite.config.ts

```diff
@@ -0,0 +1,29 @@
+import { defineConfig } from 'vite'
+import dts from 'vite-plugin-dts'
+
+import pkg from './package.json'
+
+export const getPeerDeps = pkg => Object.keys(pkg.peerDependencies || {})
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  build: {
+    lib: {
+      entry: 'src/index.ts',
+      formats: ['es', 'cjs', 'umd'],
+      name: 'esvintersection',
+      fileName: 'index',
```

Comment:
![image](https://user-images.githubusercontent.com/91891939/209924276-6432cba0-1eb0-49d9-8c1f-8e34a71fb5f7.png)
The default file endings got my eyebrow moving. Could we instead use the conventional file endings for the different formats?
```suggestion
      fileName: (format) => {
        const fileName = 'index'
        switch(format) {
          case 'es':
            return `${fileName}.mjs`
          case 'cjs':
            return `${fileName}.cjs`
          case 'umd':
            return `${fileName}.umd.js`
          default:
            return `${fileName}.js`
        }
      },
```
```package.json
  "exports": {
    "import": "./dist/index.mjs",
    "require": "./dist/index.cjs",
    "browser": "./dist/index.umd.js"
  },
```